src/features/Chat/components/ActiveWindow/index.tsx

```diff
@@ -0,0 +1,234 @@
+        <HeaderBar>
+          <ProfileInfo>
+            {getProfileIcon(chat.status)}
+            <MentorName variant="h2">{chat.displayName}</MentorName>
+            <MentorBio variant="p">{chat.status}</MentorBio>
+          </ProfileInfo>
+          {showSearch ? (
+            <SearchBar>
+              <SearchInput
+                variant="iconInput"
+                color={searchValue ? 'blueDark' : 'greyFaded'}
+                leftIcon={{
+                  sizeInPx: searchInputIconSize,
+                  variant: 'search',
+                }}
+                rightButton={{
+                  onClick: () => setShowSearch(false),
+                  sizeInPx: closeInputIconSize,
+                  variant: 'closeWithBackground',
+                }}
+                onChange={setSearchValue}
+                placeholder={t('header.search')}
+                value={searchValue}
+              />
+            </SearchBar>
+          ) : (
+            <Buttons>
+              <IconButton
+                variant="search"
+                sizeInPx={24}
+                onClick={() => setShowSearch(true)}
+              />
+              {chat.status === 'ok' ? (
+                <>
+                  <StatusButton
+                    onClick={() => updateStatus('archived')}
+                    icon="archive"
+                    text={t('header.archive')}
+                  />
+                  <StatusButton
+                    onClick={() => updateStatus('banned')}
+                    icon="block"
+                    text={t('header.block')}
+                  />
+                </>
+              ) : (
+                <StatusButton
+                  onClick={() => updateStatus('ok')}
+                  icon="return"
+                  text={t('header.restore')}
+                />
+              )}
+            </Buttons>
+          )}
+        </HeaderBar>
+        <MessageList
+          messageList={chat.messages}
+          buddyId={chat.buddyId}
+          status={chat.status}
+          isLoading={isLoading}
+        />
+        <MessageField>
+          <Input
+            variant="textarea"
+            color={text ? 'blueDark' : 'greyFaded'}
+            onChange={setText}
+            placeholder={t('input.placeholder')}
+            value={text}
+          />
+          <SendButton
+            variant="send"
+            sizeInPx={46}
+            onClick={() => handleMessageSend(chat.buddyId, text)}
+          />
+        </MessageField>
+      </ActiveChatContainer>
+    )
+  );
+};
+
+const ActiveChatContainer = styled.div`
+  background-color: ${palette.white};
+  border-radius: 10px;
+  display: flex;
+  flex: 1 1 auto;
+  flex-direction: column;
+  height: 100%;
+`;
+
+const HeaderBar = styled.div`
+  border-bottom: 1px solid ${palette.greyLight};
+  box-sizing: border-box;
+  display: flex;
+  gap: 30px;
+  height: 80px;
+  justify-content: space-between;
+  min-width: 800px;
+  padding: 14px 40px;
+`;
```

Comment:
I would abstract everything inside the Headerbar in an own component, create a folder in 

src/features/Chat/components/ActiveWindow/components/Headerbar

then have index-file that returns this:

```typescript
        <HeaderBar>
          <ProfileInfo>
            {getProfileIcon(chat.status)}
            <MentorName variant="h2">{chat.displayName}</MentorName>
            <MentorBio variant="p">{chat.status}</MentorBio>
          </ProfileInfo>
          {showSearch ? (
            <SearchBar>
              <SearchInput
                variant="iconInput"
                color={searchValue ? 'blueDark' : 'greyFaded'}
                leftIcon={{
                  sizeInPx: searchInputIconSize,
                  variant: 'search',
                }}
                rightButton={{
                  onClick: () => setShowSearch(false),
                  sizeInPx: closeInputIconSize,
                  variant: 'closeWithBackground',
                }}
                onChange={setSearchValue}
                placeholder={t('header.search')}
                value={searchValue}
              />
            </SearchBar>
          ) : (
            <Buttons>
              <IconButton
                variant="search"
                sizeInPx={24}
                onClick={() => setShowSearch(true)}
              />
              {chat.status === 'ok' ? (
                <>
                  <StatusButton
                    onClick={() => updateStatus('archived')}
                    icon="archive"
                    text={t('header.archive')}
                  />
                  <StatusButton
                    onClick={() => updateStatus('banned')}
                    icon="block"
                    text={t('header.block')}
                  />
                </>
              ) : (
                <StatusButton
                  onClick={() => updateStatus('ok')}
                  icon="return"
                  text={t('header.restore')}
                />
              )}
            </Buttons>
          )}
        </HeaderBar>
````

Then continue abstracting the other conditional returns to their own components, so in the end its something like this:

```typescript
<Header>
<Info {...infoProps}/>
<Search {...searchProps}/>
<Actions {...actionProps}/>
</Header>
```

I think this will improve the readability of this component a lot. The point is that you should by a glance get an understanding what is happening.

Reply:
Yes this is defenitely the goal! I'll work on this some more!